server/src/graphql/resolvers/pathway.js

```diff
@@ -1,56 +1,89 @@
-const queries = require('../../db/queries/queries')
+const queries = require("../../db/queries/queries")
+const github = require("../../utils/github")
 
 const resolver = {
     Mutation: {
-        async createUpdatePathway(_, {id, name, steps}) {
+        async createUpdatePathway(_, { id, name, steps }) {
             const query = queries.pathway.createUpdatePathway({
-                id, name, steps
+                id,
+                name,
+                steps,
             })
 
             try {
                 await query.run()
-                return { status: 'OK', message: null }
+                return { status: "OK", message: null }
             } catch (e) {
-                return { status: 'ERROR', message: e.toString() }
+                return { status: "ERROR", message: e.toString() }
             }
         },
-        async deletePathway(_, {id, steps, whole}) {
+        async deletePathway(_, { id, steps, whole }) {
             const query = queries.pathway.deletePathway({
-                id, steps, whole
+                id,
+                steps,
+                whole,
             })
 
             try {
                 await query.run()
-                return { status: 'OK', message: null }
+                return { status: "OK", message: null }
             } catch (e) {
-                return { status: 'ERROR', message: e.toString() }
+                return { status: "ERROR", message: e.toString() }
             }
         },
-        async createUpdateContent(_, {id, title, content}) {
+        async createUpdateContent(_, { id, title, content }, context) {
             const query = queries.content.createUpdateContent({
-                id, title, content 
+                id,
+                title,
+                content,
             })
 
             try {
                 await query.run()
-                return { status: 'OK', message: null}
+
+                // commit changes to github in the background
+                // slugify id to create file name so that file name
+                // remains unchanged even if title changes
+                github.githubCommit({
+                    id,
+                    content,
+                    author_name: context.user.name,
+                    author_email: context.user.email,
+                })
```

Comment:
If the GitHub commit fails here, won't the database have a different state than the repository? In this case should we retry the commit or rollback the database? We are returning an error on the mutation but the database contains the data posted in the mutation.

Reply:
Added retry and back off in 2135f89. All errors are logged and if the request still fails the changes can be committed manually - the chances of this happening are minimal.

Moreover since the last request that is update ref actually updates the HEAD to point to the new commit, so unless that is successful the main branch is never updated. This means that the GitHub repo can never be in a inconsistent state. The database is the primary source of truth while GitHub repo is behaving as the publicly accessible redundant storage with feature for collaboration.